plotting/colorscales.py

```diff
@@ -0,0 +1,94 @@
+import matplotlib.pyplot as plt
+from matplotlib.pyplot import get_cmap
+from matplotlib import cm
+import matplotlib.colors
+import seaborn as sns
+import numpy as np
+import pandas as pd
+
+black = matplotlib.colors.to_rgba("black")
+gray = matplotlib.colors.to_rgba("gray")
+white = matplotlib.colors.to_rgba("white")
+
+def _mpl_normalize(s, method, cmap='vlag', **kwargs):
+
+    '''
+    Internal function that returns mapper for mapping continuous variable to 
+    matplotlib colormap
+
+    s - pandas Series to be mapped
+    cmap - String name of acceptable matplotlib colormap
+    method - Normalization function for normalizing data to [0,1] scale
+    **kwargs - args for normalization function
+
+    Returns pandas Series object
+    '''
+
+    # Create mapper using desired method and colormap
+    mapper = cm.ScalarMappable(method(**kwargs), cmap=get_cmap(cmap))
+
+    return((s.apply(lambda x: mapper.to_rgba(x)),))
+
+def _map_categorical_colors(s, values, dtype = 'categorical'):
+    if dtype == 'categorical':
+        colorscheme = sns.color_palette("bright")+sns.color_palette("pastel")
+    elif dtype == 'binary':
+        colorscheme = [black, gray, white]
+
+    color_dict=dict(zip(values, color_scheme))
+    
+
+    #Return color key as well for referencing
+    return((s.map(color_dict), color_dict))
```

Comment:
I would use an enum here so you can use a switch case safely.
```suggestion
from enum import Enum

class DataColorStyle(Enum):
    CATEGORICAL = 1
    BINARY = 2

def _map_categorical_colors(s, values, dtype = DataColorStyle.CATEGORICAL):
    match dtype:
         case DataColorStyle.CATEGORICAL:
              colorscheme = sns.color_palette("bright")+sns.color_palette("pastel")
         case DataColorStyle.BINARY:
              colorscheme = [black, gray, white]
              
    color_dict=dict(zip(values, color_scheme))
    

    #Return color key as well for referencing
    return((s.map(color_dict), color_dict))
```